spyder.js

```diff
@@ -125,57 +119,122 @@ class SpyderInstance extends InstanceBase {
 										for (let i = 0; i < c; i++) {
 											let id = parseInt(r[i * 2])
 											let name = decodeURIComponent(r[i * 2 + 1])
-											let rr = this.reg[id]
+											let rr = this.reg[-1]
 											if (rr === undefined) {
 												rr = { id, active: false, scriptId: null }
-												this.variableDefs.push({ name: `Register ${id + 1} name`, variableId: `r_name_${id + 1}` })
+												this.variableDefs.push({ name: `Script ${id}`, variableId: `ck_${id}` })
```

Comment:
Changing previously defined variable names will break any existing buttons that use them.